File: android/src/main/java/com/thomasvidas/advertising/AdvertisingIdPlugin.java
package com.thomasvidas.advertising;

import android.content.Context;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "AdvertisingId")
public class AdvertisingIdPlugin extends Plugin {

    private AdvertisingId implementation = new AdvertisingId();

    @PluginMethod
    public void requestTracking(PluginCall call) {
        String status = implementation.getTrackingStatus(getContext());

        JSObject output = new JSObject();
        output.put("value", status);
        call.resolve(output);
    }

    @PluginMethod
    public void getAdvertisingId(PluginCall call) {
        Context context = getContext();
        String id = implementation.getTrackingId(context);
        String status = implementation.getTrackingStatus(context);

        JSObject output = new JSObject();
        output.put("id", id);
        output.put("status", status);
        call.resolve(output);
    }

    @PluginMethod
    public void getAdvertisingStatus(PluginCall call) {
        String status = implementation.getTrackingStatus(getContext());

        JSObject output = new JSObject();
        output.put("status", status);
        call.resolve(output);
    }
}

File: ios/Plugin/AdvertisingIdPlugin.swift
import Foundation
import Capacitor

@objc(AdvertisingIdPlugin)
public class AdvertisingIdPlugin: CAPPlugin {
    private let implementation = AdvertisingId()

    @objc func requestTracking(_ call: CAPPluginCall) {
        implementation.requestTracking { status in
            call.resolve([ "value": status.name() ])
        }
    }

    @objc func getAdvertisingId(_ call: CAPPluginCall) {
        let id = implementation.getAdvertisingId()
        let status = implementation.getAdvertisingStatus()
        call.resolve([
            "id": id,
            "status": status.name()
        ])
    }

    @objc func getAdvertisingStatus(_ call: CAPPluginCall) {
        let status = implementation.getAdvertisingStatus()
        call.resolve([
            "status": status.name()
        ])
    }
}
